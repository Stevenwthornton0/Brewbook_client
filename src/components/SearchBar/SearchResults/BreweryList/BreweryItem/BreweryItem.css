section {
    color: black;
}

section > a {
    color: black;
    margin-top: 0px;
    font-size: 20px;
}

.breweryItem {
    margin-top: 30px;
    border: 2px solid #424242;
    border-radius: 4px;
    padding: 30px 40px;
    padding-bottom: 50px;
}

.breweryItem > h3 {
    margin-top: 0px;
    font-size: 22px;
}

.breweryItem p {
    margin-top: 8px;
    margin-bottom: 0px;
}

.container1 {
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.container3 {
    text-align: end;
}


@media screen and (max-width: 500px) {
    .breweryItem {
        padding: 20px 30px;
        padding-bottom: 30px;
    }

    .container1 {
        display: flex;
        flex-direction: column;
    }

    .container3 {
        text-align: start;
    }

    .url {
        margin-left: 10px;
    }
}
