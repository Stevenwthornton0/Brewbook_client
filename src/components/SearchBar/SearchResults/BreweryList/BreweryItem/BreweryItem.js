import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { displayPhone, displayWebsite } from '../../../../../Utils/Utils';

class BreweryItem extends Component {


    render() {
        let { brewery } = this.props;
        return (
            <Link to={`/brewery/${brewery.id}`}>
                <section className='breweryItem'>
                    <h3>{brewery.name}</h3>
                    <p>{brewery.street}</p>
                    <p>{brewery.city}, {brewery.state}</p>
                    {displayPhone(brewery)}
                    {displayWebsite(brewery)}
                    <br></br>
                </section>
            </Link>
        )
    }
}

export default BreweryItem;