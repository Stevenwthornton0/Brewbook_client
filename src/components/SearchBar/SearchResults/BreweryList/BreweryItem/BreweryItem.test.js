import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import BreweryItem from './BreweryItem';

Enzyme.configure({ adapter: new Adapter() })

describe(`BreweryItem smoke test`, () => {
    it('BreweryItem renders without crashing', () => {
        const brewery = {
            id: 6780,
            name: "Jester King Brewery",
            brewery_type: "micro",
            street: "13187 Fitzhugh Rd",
            city: "Austin",
            state: "Texas",
            postal_code: "78736-6510",
            country: "United States",
            longitude: "-98.0824692",
            latitude: "30.2547264",
            phone: "5125375100",
            website_url: "http://www.jesterkingbrewery.com",
            updated_at: "2018-08-24T16:21:26.875Z"
            }
        const wrapper = shallow(<BreweryItem brewery={brewery}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})