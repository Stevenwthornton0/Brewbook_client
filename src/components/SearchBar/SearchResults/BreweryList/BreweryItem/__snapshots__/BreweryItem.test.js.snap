// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreweryItem smoke test BreweryItem renders without crashing 1`] = `
<Link
  className="brewery"
  to="/brewery/6780"
>
  <section
    className="breweryItem"
  >
    <h3>
      Jester King Brewery
    </h3>
    <div
      className="container1"
    >
      <div
        className="container2"
      >
        <p>
          13187 Fitzhugh Rd
        </p>
        <p>
          Austin
          , 
          Texas
        </p>
      </div>
      <div
        className="container3"
      >
        <p
          className="brewery website"
        >
          <FontAwesomeIcon
            border={false}
            className="logo"
            fixedWidth={false}
            flip={null}
            icon="mouse"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
          <span
            className="url"
          >
            jesterkingbrewery.com
          </span>
        </p>
        <p
          className="brewery phone"
        >
          <FontAwesomeIcon
            border={false}
            className="logo"
            fixedWidth={false}
            flip={null}
            icon="phone"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
           
          512-537-5100
        </p>
      </div>
    </div>
  </section>
</Link>
`;
